client/helpers/scenes.ts

```diff
@@ -337,8 +332,46 @@ export async function createSceneAndSceneItemsEXPERIMENTAL({
   let batchQueuedOps: QueuedBatchOps[] = [];
   let batchErrors: QueuedBatchOps[] = [];
   let sceneItemErrors: SceneItemError[] = [];
-  for (let depth = 0; depth < createSceneItemReqs.length; depth++) {
-    const createItemReqs: CreateSceneItemRequest[] = createSceneItemReqs[depth];
+
+  const reqMap: Map<string, CreateSceneItemRequest[]> = new Map();
+  let nextChildren: CreateSceneItemRequest[] = [];
+  reqMap.set('', nextChildren);
+
+  // create parent map and set ordinals based on request order
+  createSceneItemReqs.forEach((req) => {
+    const reqParent =
+      req.data.attributes.parent ??
+      req.data.relationships.parent?.data.id ??
+      '';
+    if (!reqMap.has(reqParent)) {
+      reqMap.set(reqParent, []);
+    }
+    const siblings = reqMap.get(reqParent);
+    if (req.data.attributes.ordinal === undefined) {
+      req.data.attributes.ordinal = siblings?.length;
+    }
+    siblings?.push(req);
+  });
+
+  // sort all scene item requests into depth sorted array of arrays
+  const depthSortedItems: CreateSceneItemRequest[][] = [];
+  while (nextChildren.length) {
```

Comment:
I think there's a bug here. The first pass of `nextChildren.length` will yield `0`. Also, generally you do not want rely on type coercion in JS, better to do `while (nextChildren.length > 0)`.